content/redirect.js

```diff
@@ -37,13 +37,37 @@ const resolve = (url) => {
         continue;
       }
 
-      const redirectPairs = fs
-        .readFileSync(redirectsFilePath, "utf-8")
+      const content = fs.readFileSync(redirectsFilePath, "utf-8");
+
+      const redirectPairs = content
         .split("\n")
         .slice(1, -1)
         .map((line) => line.trim().split(/\s+/));
+
+      const normalizedPairs = new Map();
+      for (const [from, to] of redirectPairs) {
+        normalizedPairs.set(from.toLowerCase(), to.toLowerCase());
+      }
+
       for (const [from, to] of redirectPairs) {
-        redirects.set(from.toLowerCase(), to);
+        let toNormalized = to.toLowerCase();
+        if (normalizedPairs.has(toNormalized)) {
```

Comment:
Thing is, I'd rather tackle this "further up". But it would be nice as "a last line of defense" that we don't have redirects from "A to B" and "B to C" when it just needs "A to C" and "B to C". After all, our Yari builder doesn't really follow redirects. If you do `Redirects.resolve("A")` you get "B" which is kinda wrong.